src/theory/arith/nonlinear_extension.cpp

```diff
@@ -2895,53 +2910,311 @@ std::vector<Node> NonlinearExtension::checkTranscendentalMonotonic() {
       }
     }
   }
-  
-  
-  
-  
   return lemmas;
 }
-  
-  
-Node NonlinearExtension::getTaylor( Node tf, Node x, unsigned n, std::vector< Node >& lemmas ) {
-  Node i_exp_base_term = Rewriter::rewrite( NodeManager::currentNM()->mkNode( kind::MINUS, x, tf[0] ) );
-  Node i_exp_base = getFactorSkolem( i_exp_base_term, lemmas );
-  Node i_derv = tf;
-  Node i_fact = d_one;
-  Node i_exp = d_one;
-  int i_derv_status = 0;
-  unsigned counter = 0;
-  std::vector< Node > sum;
-  do {
-    counter++;
-    if( tf.getKind()==kind::EXPONENTIAL ){
-      //unchanged
-    }else if( tf.getKind()==kind::SINE ){
-      if( i_derv_status%2==1 ){
-        Node arg = NodeManager::currentNM()->mkNode( kind::MINUS, 
-                     NodeManager::currentNM()->mkNode( kind::MULT, 
-                       NodeManager::currentNM()->mkConst( Rational(1)/Rational(2) ),
-                       NodeManager::currentNM()->mkNullaryOperator( NodeManager::currentNM()->realType(), kind::PI ) ),
-                     tf[0] );
-        i_derv = NodeManager::currentNM()->mkNode( kind::SINE, arg );
+
+int NonlinearExtension::regionToMonotonicityDir(Kind k, int region)
+{
+  if (k == kind::EXPONENTIAL)
+  {
+    if (region == 1)
+    {
+      return 1;
+    }
+  }
+  else if (k == kind::SINE)
+  {
+    if (region == 1 || region == 4)
+    {
+      return -1;
+    }
+    else if (region == 2 || region == 3)
```

Comment:
ditto

Reply:
I think since this else doesn't add any space, I'd prefer keeping it

Reply:
same goes for those below